components/telemetry-operator/internal/webhook/logparser/mocks/dryrun.go

```diff
@@ -1,25 +1,27 @@
-// Code generated by mockery v2.13.1. DO NOT EDIT.
+// Code generated by mockery v2.14.0. DO NOT EDIT.
 
 package mocks
 
 import (
 	context "context"
 
 	mock "github.com/stretchr/testify/mock"
+
+	v1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
 )
 
 // DryRunner is an autogenerated mock type for the DryRunner type
 type DryRunner struct {
 	mock.Mock
 }
 
-// RunParser provides a mock function with given fields: ctx, configFilePath
-func (_m *DryRunner) RunParser(ctx context.Context, configFilePath string) error {
-	ret := _m.Called(ctx, configFilePath)
+// DryRunParser provides a mock function with given fields: ctx, parser
+func (_m *DryRunner) DryRunParser(ctx context.Context, parser *v1alpha1.LogParser) error {
```

Comment:
We have the same mock in the logparser and logpipeline package. Is that intended? I'm wondering if we should move it to dryrun/mocks.

Reply:
Good catch 😅 forgot to run mockery. The interface has been split and is provided by the consumer now, in this case each webhook. Our DryRunner provides the implementations for both interfaces.

Reply:
Due to the fact that interfaces in go are implemented implicitly, usually, you would implement packages in a way that concrete structs are returned: https://github.com/golang/go/wiki/CodeReviewComments#interfaces. So interfaces are usually defined by consumers, not producers. In this case, for example, the `DryRunner` has 2 methods. But every webhook only needs one of them, so it can define an interface that only consists of what is needed. This approach allows you to write more loosely-coupled packages.

I believe, we should use the same approach in other packages as well.